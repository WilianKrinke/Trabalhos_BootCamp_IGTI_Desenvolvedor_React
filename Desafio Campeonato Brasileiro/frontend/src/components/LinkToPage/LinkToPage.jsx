import React from 'react';
import { string } from 'prop-types';
import { Link } from 'react-router-dom';

const LinkToPage = ({url, title}) => {
    return <Link to={url} data-testid="routerlink">{title}</Link>
}

LinkToPage.propTypes = {
    url: string.isRequired,
    title: string.isRequired
}

export default LinkToPage;
