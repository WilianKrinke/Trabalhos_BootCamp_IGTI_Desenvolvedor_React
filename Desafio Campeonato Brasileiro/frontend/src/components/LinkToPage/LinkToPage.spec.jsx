import React from "react";
import LinkToPage from "./LinkToPage";
import renderer from 'react-test-renderer'
import {render, screen} from '@testing-library/react'
import { BrowserRouter } from "react-router-dom";

describe('LinkToPage test component', () => {
    it('should render component with props "url=/secondpage" and "title=Segunda Página"', () => {

        const {getByTestId} = render(
                <BrowserRouter>
                    <LinkToPage url="/secondpage" title="Segunda Página"/>
                </BrowserRouter>
            )
        
        expect(getByTestId('routerlink')).toHaveTextContent("Segunda Página")        
        expect(screen.getByRole('link')).toHaveAttribute('href', '/secondpage');  
    });
    
    it('should render component with props "url=/thirdpage" and "title=Terceira Página"', () => {

        const {getByTestId} = render(
                <BrowserRouter>
                    <LinkToPage url="/thirdpage" title="Terceira Página"/>
                </BrowserRouter>
            )
        
        expect(getByTestId('routerlink')).toHaveTextContent("Terceira Página")        
        expect(screen.getByRole('link')).toHaveAttribute('href', '/thirdpage');  
    });
    
    it('matches snapshot 1', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <LinkToPage url="/secondpage" title="Segunda Página"/>
            </BrowserRouter>
        ).toJSON()
        expect(tree).toMatchSnapshot();
    });

    it('matches snapshot 2', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <LinkToPage url="/thirdpage" title="Terceira Página"/>
            </BrowserRouter>
        ).toJSON()
        expect(tree).toMatchSnapshot();
    });
    
});
