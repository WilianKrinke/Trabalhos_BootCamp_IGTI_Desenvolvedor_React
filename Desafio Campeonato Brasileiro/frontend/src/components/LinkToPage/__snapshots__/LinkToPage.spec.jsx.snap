// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkToPage component matches snapshot 1 1`] = `
<a
  data-testid="routerlink"
  href="/secondpage"
  onClick={[Function]}
>
  Segunda P치gina
</a>
`;

exports[`LinkToPage component matches snapshot 2 1`] = `
<a
  data-testid="routerlink"
  href="/thirdpage"
  onClick={[Function]}
>
  Terceira P치gina
</a>
`;

exports[`LinkToPage test component matches snapshot 1 1`] = `
<a
  data-testid="routerlink"
  href="/secondpage"
  onClick={[Function]}
>
  Segunda P치gina
</a>
`;

exports[`LinkToPage test component matches snapshot 2 1`] = `
<a
  data-testid="routerlink"
  href="/thirdpage"
  onClick={[Function]}
>
  Terceira P치gina
</a>
`;
